lib-ee/emqx_ee_connector/test/ee_connector_clickhouse_SUITE.erl

```diff
@@ -0,0 +1,199 @@
+        {_, QueryNoParamsRes} = QueryNoParamsResWrapper,
+        ?assertMatch(<<"1">>, string:trim(QueryNoParamsRes))
+    % Driver currently has no support for query with parms
+    % ?assertMatch({ok, _, [{1}]}, emqx_resource:query(PoolName, test_query_with_params())),
+    end)(),
+    ?assertEqual(ok, emqx_resource:stop(PoolName)),
+    % Resource will be listed still, but state will be changed and healthcheck will fail
+    % as the worker no longer exists.
+    {ok, ?CONNECTOR_RESOURCE_GROUP, #{
+        state := State,
+        status := StoppedStatus
+    }} =
+        emqx_resource:get_instance(PoolName),
+    ?assertEqual(stopped, StoppedStatus),
+    ?assertEqual({error, resource_is_stopped}, emqx_resource:health_check(PoolName)),
+    % Resource healthcheck shortcuts things by checking ets. Go deeper by checking pool itself.
+    ?assertEqual({error, not_found}, ecpool:stop_sup_pool(ReturnedPoolName)),
+    % Can call stop/1 again on an already stopped instance
+    ?assertEqual(ok, emqx_resource:stop(PoolName)),
+    % Make sure it can be restarted and the healthchecks and queries work properly
+    ?assertEqual(ok, emqx_resource:restart(PoolName)),
+    % async restart, need to wait resource
+    timer:sleep(500),
+    {ok, ?CONNECTOR_RESOURCE_GROUP, #{status := InitialStatus}} =
+        emqx_resource:get_instance(PoolName),
+    ?assertEqual({ok, connected}, emqx_resource:health_check(PoolName)),
+    (fun() ->
+        QueryNoParamsResWrapper =
+            emqx_resource:query(PoolName, test_query_no_params()),
+        ?assertMatch({ok, _}, QueryNoParamsResWrapper),
+        {_, QueryNoParamsRes} = QueryNoParamsResWrapper,
+        ?assertMatch(<<"1">>, string:trim(QueryNoParamsRes))
+    % Driver currently has no support for query with parms
+    % ?assertMatch({ok, _, [{1}]}, label2, emqx_resource:query(PoolName, test_query_with_params())),
+    end)(),
+    % Stop and remove the resource in one go.
+    ?assertEqual(ok, emqx_resource:remove_local(PoolName)),
+    ?assertEqual({error, not_found}, ecpool:stop_sup_pool(ReturnedPoolName)),
+    % Should not even be able to get the resource data out of ets now unlike just stopping.
+    ?assertEqual({error, not_found}, emqx_resource:get_instance(PoolName)).
+
+% %%------------------------------------------------------------------------------
+% %% Helpers
+% %%------------------------------------------------------------------------------
+
+clickhouse_config() ->
+    RawConfig = list_to_binary(
+        io_lib:format(
+            ""
+            "\n"
+            "    auto_reconnect = true\n"
+            "    database = mqtt\n"
+            "    username= default\n"
+            "    password = public\n"
+            "    pool_size = 8\n"
+            "    url = \"http://~s:~b\"\n"
+            "    "
+            "",
+            [?CLICKHOUSE_HOST, ?CLICKHOUSE_DEFAULT_PORT]
+        )
+    ),
+
+    {ok, Config} = hocon:binary(RawConfig),
```

Comment:
Suggestion: since this does not check against any schema, creating a map directly could be simpler to read and modify.

Reply:
Thanks, fixed